astropy/modeling/rotations.py

```diff
@@ -187,9 +187,6 @@ def evaluate(cls, x, y, angle):
         single coordinate pair.
         """
 
-        if x.shape != y.shape:
-            raise ValueError("Expected input arrays to have the same shape")
```

Comment:
Let's not make this change, since I agree as you wrote there's a deeper problem there with how scalar outputs are being treated by mapping.